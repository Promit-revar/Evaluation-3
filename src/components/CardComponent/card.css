.card{
    display: flex;
    flex-wrap: wrap;
    justify-content: space-around;
    padding: 120px;
    gap: 50px;
}