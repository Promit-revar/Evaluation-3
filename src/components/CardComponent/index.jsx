/* eslint-disable */
import React from 'react';
import makeRequest from '../../utils/makeRequest';
import { eventsContext } from '../../contexts/dataEventContext';
import ExpandedCard from '../ExpandedCard';
import getEvents from '../../endpoints/getEvents';
import EventCardComponent from '../EventCardComponent/index';
import './card.css';
import SearchFilterComponent from '../searchAndFilterComponent';
export default function CardComponent() {
  const { events, setEvents} = React.useContext(eventsContext);
  const [filteredEvents, setFilteredEvents] = React.useState([]);
  const [card, setCard] = React.useState({});
  React.useEffect(() => {
    makeRequest(getEvents).then(response => {
      console.log("response",response);
      setEvents([...response]);
      setFilteredEvents([...response]);
    }).catch(error => console.log(error));
  }, []);
  const expandCard = (event) => {
    console.log(event);
    setCard(event);
  }
  const handleSearch = (searchText) => {
    setFilteredEvents([...events.filter((data) => data.name.search(searchText) != -1 || data.description.search(searchText) != -1)]);
  }
  if(Object.keys(card).length === 0){
  return (
    <div>
        <SearchFilterComponent searchFunction={handleSearch}/>
        <div className='card'>
            {console.log(filteredEvents)}
        {filteredEvents.map(event => (
            <EventCardComponent key={event.id} event={event} viewCard={expandCard}/>
        ))}
        </div>
    </div>
  );
    }
    else{
    return (
        <div className='card'>
            <ExpandedCard details={card} />
        </div>
    );
    }

};