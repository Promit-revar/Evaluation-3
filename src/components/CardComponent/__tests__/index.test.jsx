import CardComponent from "..";
import React from "react";
import { render} from "@testing-library/react";
describe("CardComponent", () => {
  it("should render without crashing", () => {
    const {asFragment}=render(<CardComponent />);
    expect(asFragment()).toMatchSnapshot();
  });
});
