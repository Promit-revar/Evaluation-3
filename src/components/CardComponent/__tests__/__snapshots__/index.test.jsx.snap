// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardComponent should render without crashing 1`] = `
<DocumentFragment>
  <div>
    <div
      class="search-filter"
    >
      <div
        class="left"
      >
        <div
          class="filter"
        >
          <svg
            aria-hidden="true"
            class="svg-inline--fa fa-filter "
            data-icon="filter"
            data-prefix="fas"
            focusable="false"
            role="img"
            viewBox="0 0 512 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M3.9 54.9C10.5 40.9 24.5 32 40 32H472c15.5 0 29.5 8.9 36.1 22.9s4.6 30.5-5.2 42.5L320 320.9V448c0 12.1-6.8 23.2-17.7 28.6s-23.8 4.3-33.5-3l-64-48c-8.1-6-12.8-15.5-12.8-25.6V320.9L9 97.3C-.7 85.4-2.8 68.8 3.9 54.9z"
              fill="currentColor"
            />
          </svg>
          <p>
             Filter
          </p>
          <svg
            aria-hidden="true"
            class="svg-inline--fa fa-chevron-up "
            data-icon="chevron-up"
            data-prefix="fas"
            focusable="false"
            role="img"
            viewBox="0 0 448 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M201.4 105.4c12.5-12.5 32.8-12.5 45.3 0l192 192c12.5 12.5 12.5 32.8 0 45.3s-32.8 12.5-45.3 0L224 173.3 54.6 342.6c-12.5 12.5-32.8 12.5-45.3 0s-12.5-32.8 0-45.3l192-192z"
              fill="currentColor"
            />
          </svg>
        </div>
        <div
          class="filter-type"
        >
          <svg
            aria-hidden="true"
            class="svg-inline--fa fa-circle-dot "
            data-icon="circle-dot"
            data-prefix="fas"
            focusable="false"
            role="img"
            viewBox="0 0 512 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M256 512c141.4 0 256-114.6 256-256S397.4 0 256 0S0 114.6 0 256S114.6 512 256 512zm0-160c-53 0-96-43-96-96s43-96 96-96s96 43 96 96s-43 96-96 96z"
              fill="currentColor"
            />
          </svg>
          <p>
            All
          </p>
        </div>
        <div
          class="reg"
        >
          <svg
            aria-hidden="true"
            class="svg-inline--fa fa-circle "
            data-icon="circle"
            data-prefix="fas"
            focusable="false"
            role="img"
            viewBox="0 0 512 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M256 512c141.4 0 256-114.6 256-256S397.4 0 256 0S0 114.6 0 256S114.6 512 256 512z"
              fill="currentColor"
            />
          </svg>
          <p>
            Seat Available
          </p>
        </div>
      </div>
      <div
        class="right"
      >
        <div
          class="search"
        >
          <input
            placeholder="Search"
            type="text"
          />
          <svg
            aria-hidden="true"
            class="svg-inline--fa fa-magnifying-glass "
            data-icon="magnifying-glass"
            data-prefix="fas"
            focusable="false"
            role="img"
            viewBox="0 0 512 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M416 208c0 45.9-14.9 88.3-40 122.7L502.6 457.4c12.5 12.5 12.5 32.8 0 45.3s-32.8 12.5-45.3 0L330.7 376c-34.4 25.2-76.8 40-122.7 40C93.1 416 0 322.9 0 208S93.1 0 208 0S416 93.1 416 208zM208 352a144 144 0 1 0 0-288 144 144 0 1 0 0 288z"
              fill="currentColor"
            />
          </svg>
        </div>
        <div
          class="reg"
        >
          <svg
            aria-hidden="true"
            class="svg-inline--fa fa-circle "
            data-icon="circle"
            data-prefix="fas"
            focusable="false"
            role="img"
            viewBox="0 0 512 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M256 512c141.4 0 256-114.6 256-256S397.4 0 256 0S0 114.6 0 256S114.6 512 256 512z"
              fill="currentColor"
            />
          </svg>
          <p>
            Registered
          </p>
        </div>
        <div
          class="reg"
        >
          <svg
            aria-hidden="true"
            class="svg-inline--fa fa-circle "
            data-icon="circle"
            data-prefix="fas"
            focusable="false"
            role="img"
            viewBox="0 0 512 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M256 512c141.4 0 256-114.6 256-256S397.4 0 256 0S0 114.6 0 256S114.6 512 256 512z"
              fill="currentColor"
            />
          </svg>
          <p>
            Bookmarked
          </p>
        </div>
      </div>
      <div
        class="radio"
      />
    </div>
    <div
      class="card"
    />
  </div>
</DocumentFragment>
`;
